test/controllers/contactPreference/ChangeEmailControllerSpec.scala

```diff
@@ -16,76 +16,290 @@
 
 package controllers.contactPreference
 
+import assets.CustomerInfoConstants.fullCustomerInfoModel
 import common.SessionKeys
+import connectors.httpParsers.GetCustomerInfoHttpParser.GetCustomerInfoResponse
 import controllers.ControllerBaseSpec
-import play.api.mvc.AnyContentAsEmpty
-import play.api.test.FakeRequest
+import mocks.MockVatSubscriptionService
+import models.errors.ErrorModel
+import org.jsoup.Jsoup
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{never, verify, when}
+import play.api.http.Status
 import play.api.test.Helpers._
-import models.contactPreferences.ContactPreference._
+import uk.gov.hmrc.http.HeaderCarrier
+import views.html.email.CaptureEmailView
 
-class ChangeEmailControllerSpec extends ControllerBaseSpec {
+import scala.concurrent.{ExecutionContext, Future}
 
-  lazy val controller = new ChangeEmailController()
-  lazy val requestWithSession: FakeRequest[AnyContentAsEmpty.type] = request.withSession((SessionKeys.currentContactPrefKey -> paper))
+class ChangeEmailControllerSpec extends ControllerBaseSpec with MockVatSubscriptionService {
 
-  "ChangeEmailController .show" when {
+  val testValidationEmail: String = "validation@example.com"
+  val testValidEmail: String = "pepsimac@gmail.com"
+  val testInvalidEmail: String = "invalidEmail"
+  val view: CaptureEmailView = injector.instanceOf[CaptureEmailView]
 
-    "user is authorised" when {
 
-      "the user currently has a digital preference" when {
+  def target() :ChangeEmailController = {
 
-        "the letterToConfirmedEmailEnabled feature switch is off" should {
+    new ChangeEmailController(
+      mockVatSubscriptionService,
+      mockErrorHandler,
+      mockAuditingService,
+      view
+    )
+  }
+
+  "Calling the show action" when {
```

Comment:
These tests are now fine but as mentioned above, now that the controller actions are moving out into `CaptureEmailController`, these would also need to be moved into `CaptureEmailControllerSpec`